_posts/2023-05-31-cloud-kickstart-components.md

```diff
@@ -0,0 +1,182 @@
+
+<img alt="architecture" src="{{ 'img/2023-05-31-cloud-kickstart-components/img.png' | prepend: site.baseurl }}"  style="margin:0px auto; max-width: 750px;">
+
+# Docker
+
+Docker Hub serves as a central repository that allows us to store, manage, and distribute our Docker images, while AWS provides the ideal platform for deploying and running our containerized applications. 
+To begin the deployment process, we build and package our application code into a Docker image, encapsulating all the necessary dependencies and configurations. 
+This Docker image acts as a self-contained unit, ensuring consistent deployment across various environments. We leverage the power of Docker Hub, a trusted and widely used container registry, to store and manage our Docker images. 
+By utilizing Docker Hub, we can easily version our images, making it simple to track and manage changes over time. 
+This ensures that our Docker images are always up to date, incorporating the latest changes and enhancements. Once our Docker images are prepared on Docker Hub, we are ready to deploy them to AWS.
+
+<img alt="architecture" src="{{ 'img/2023-05-31-cloud-kickstart-components/img_1.png' | prepend: site.baseurl }}"  style="margin:0px auto; max-width: 750px;">
+
+# GitHub Actions
+
+We use the capabilities of GitHub Actions, a workflow automation tool. With GitHub Actions, we have automated various tasks, including deploying to Docker and seamlessly integrating our applications with multiple Amazon Web Services (AWS).
+
+<img alt="architecture" src="{{ 'img/2023-05-31-cloud-kickstart-components/img_2.png' | prepend: site.baseurl }}"  style="margin:0px auto; max-width: 750px;">
+
+One of the key benefits of GitHub Actions is its ability to automate the deployment of applications using Docker images. Using a simple configuration, we have set up workflows that automatically build our applications, package them into Docker images, and push those images to a Docker registry. 
+This automation saves us valuable time and effort, ensuring that our applications are always up to date and readily available for deployment.
+
+This automation allows us to seamlessly deploy our applications, configure the necessary settings, and utilize the full capabilities of AWS without manual intervention. For example, when triggering a deployment workflow, GitHub Actions pulls the Docker image from the registry, distributes it to ECR and deploys it to an Elastic Kubernetes Service (EKS) cluster . Simultaneously, it can create S3 buckets for storage, create/add some CloudWatch metrics, set up the alarms and necessary permissions and configurations, all in an automated and reliable manner. 
```

Comment:
```suggestion
This automation allows us to seamlessly deploy our applications, configure the necessary settings, and utilize the full capabilities of AWS without manual intervention. For example, when triggering a deployment workflow, GitHub Actions pulls the Docker image from the registry, distributes it to ECR and deploys it to an Elastic Kubernetes Service (EKS) cluster. Simultaneously, it can create S3 buckets for storage, create/add some CloudWatch metrics, and set up the alarms and necessary permissions and configurations, all in an automated and reliable manner.
```